hypervisor/dm/vpci/pci_pt.c

```diff
@@ -217,22 +217,35 @@ static void vdev_pt_map_mem_vbar(struct pci_vdev *vdev, uint32_t idx)
  */
 static void vdev_pt_allow_io_vbar(struct pci_vdev *vdev, uint32_t idx)
 {
-	struct pci_bar *vbar;
-	uint64_t vbar_base = get_vbar_base(vdev, idx); /* vbar (gpa) */
-
-	vbar = &vdev->bar[idx];
+	/* For SOS, all port IO access is allowed by default, so skip SOS here */
+	if (!is_sos_vm(vdev->vpci->vm)) {
+		struct pci_bar *vbar = &vdev->bar[idx];
+		uint64_t vbar_base = get_vbar_base(vdev, idx); /* vbar (gpa) */
+		if (vbar_base != 0UL) {
+			allow_guest_pio_access(vdev->vpci->vm, (uint16_t)vbar_base, (uint32_t)(vbar->size));
+			/* Remember the previously allowed IO vbar base */
+			vdev->bar_base_mapped[idx] = vbar_base;
```

Comment:
allow is allow. It never changes the data structure other than  the PIO allow used.

Too many same issues.